File: 3.x.x/crates-utopia/messaging/src/messages/email/attachment.rs
//! PHP `Utopia\Messaging\Messages\Email\Attachment`.

/// PHP `Utopia\Messaging\Messages\Email\Attachment`.
#[derive(Debug, Clone)]
pub struct Attachment {
    name: String,
    path: String,
    type_name: String,
    content: Option<Vec<u8>>,
}

impl Attachment {
    /// PHP `__construct($name, $path, $type, $content = null)`.
    ///
    /// `$content` is a PHP string; here it is raw bytes.
    #[must_use]
    pub fn new(
        name: impl Into<String>,
        path: impl Into<String>,
        type_name: impl Into<String>,
        content: Option<Vec<u8>>,
    ) -> Self {
        Self {
            name: name.into(),
            path: path.into(),
            type_name: type_name.into(),
            content,
        }
    }

    /// PHP `getName`.
    #[must_use]
    pub fn get_name(&self) -> &str {
        &self.name
    }

    /// PHP `getPath`.
    #[must_use]
    pub fn get_path(&self) -> &str {
        &self.path
    }

    /// PHP `getType`.
    #[must_use]
    pub fn get_type(&self) -> &str {
        &self.type_name
    }

    /// PHP `getContent` (raw bytes).
    #[must_use]
    pub fn get_content(&self) -> Option<&[u8]> {
        self.content.as_deref()
    }
}

File: packages/messaging/src/Utopia/Messaging/Messages/Email/Attachment.php
<?php

declare(strict_types=1);

namespace Utopia\Messaging\Messages\Email;

class Attachment
{
    /**
     * @param string $name  The name of the file.
     * @param string $path  The path of the file.
     * @param string $type  The MIME type of the file.
     * @param ?string $content  Raw string content of the file (used instead of path when non-null).
     */
    public function __construct(
        private readonly string $name,
        private readonly string $path,
        private readonly string $type,
        private readonly ?string $content = null,
    ) {}

    public function getName(): string
    {
        return $this->name;
    }

    public function getPath(): string
    {
        return $this->path;
    }

    public function getType(): string
    {
        return $this->type;
    }

    public function getContent(): ?string
    {
        return $this->content;
    }
}
